### ISSUE TITLE
Add 3D bounding box toggle button to toolbar

### ISSUE THREAD

**Issue by zacsimile** [python-microscopy/python-microscopy#236]
Optional display of a 3D bounding box.

**Comment by barentine** [comment]
I believe we have this in `PYME/LMVis/layers/ScaleBoxOverlayLayer`, though it'll probably need to be gone over/tested again as well as added back to the UI

**Comment by zacsimile** [comment]
Looks like it's loaded every time, just hidden. If you go to the shell and type `glCanvas.underlays[0]._show = True`, it pops up. Pretty neat!

**Comment by zacsimile** [comment]
Presumably we should just make a button to show/hide this and leave it hidden by default @David-Baddeley?

**Comment by barentine** [comment]
If I'm not crazy, once upon a time there was a UI to toggle whether there was a grid or just the edges, whether box sizes of each dimension were shown or only one, etc.

**Comment by barentine** [comment]
PYME/LMVis/Extras/scaleBox.py

I think Michael made this pretty complete after sitting down with Yongdeng.

**Comment by zacsimile** [comment]
Oh I see. Under `Extras>Scale Box`. I just have to update the `GridSizer` argument and we're good to go. PR to follow.

**Comment by zacsimile** [comment]
I still think it would be nice to have a button at the top of the GUI to toggle this on and off, but the controls Michael built are back.

**Comment by David-Baddeley** [comment]
Reworking the UI for this to put a show/hide toggle on the toolbar (and make it slightly less ugly) has been on the TODO list for ages.

I know you guys keep saying that everything works on wx4, but this is another example of something which doesn't. I'm pretty sure there are still lots of similar ones that we haven't found yet - we're slowly ticking them off, but this is why I want to keep wx3 compatibility for now - so I know we have a version where everything works that we can compare to.

**Comment by David-Baddeley** [comment]
To elaborate on slightly less ugly - I think the settings - colour, axis flipping etc probably belong in a settings dialog, and not something that is visible the whole time - the activating the scale box from the menu is a bit of a crummy workaround in the meantime.

**Comment by zacsimile** [comment]
It works on wx4 now! And I searched the rest of the code and I think those were the last of the deprecated `GridSizer` calls. 

I agree with your proposed UI changes.

**Comment by barentine** [comment]
@David-Baddeley I agree that there're probably similar wx4 migration errors lurking, but they're probably similarly trivial. Is it reckless to call the past releases the ones we know worked with wx3 and take a small wx4 plunge with the source?

**Comment by David-Baddeley** [comment]
Not sure whether to close this, or to mark as an enhancement (re adding to toolbar etc..)

**Comment by zacsimile** [comment]
I'd say enhancement (re-adding to toolbar).
